lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -251,14 +252,19 @@ define((require, exports, module) ->
       return token1.type == token2.type and token1.value == token2.value
 
     getEquationStart: (tokenIterator) ->
-      # if tokenIterator initialy is on the empty line, it's current token is null
+      # if tokenIterator is initialy on the empty line, its current token is null
       if not tokenIterator.getCurrentToken()?
         tokenIterator.stepForward()
-      # following cycle pushes tokenIterator to the end of
-      # beginning sequence, if it is inside one
-      for token in EquationRangeHandler.BEGIN_EQUATION_TOKEN_SEQUENCE
-        if EquationRangeHandler.equalTokens(token, tokenIterator.getCurrentToken())
-          tokenIterator.stepForward()
+        # if tokenIterator.getCurrentToken() is still null, then we're at the end of a file
+        if not tokenIterator.getCurrentToken()?
+          return null
+      else
```

Comment:
This looks like a bug because if `tokenIterator.getCurrenttoken` is null then you won't run this loop which pushes iterator forward. This is actually not a bug provided that we do not allow for empty lines in \begin{equation}, however it needs to be documented (like, "If token iterator is positioned at empty line then it is certainly not in the \begin{equation}") or just rewritten so that it didn't look like a bug (remove else, run the loop unconditionally)